rustls/examples/internal/test_ca.rs

```diff
@@ -0,0 +1,365 @@
+use std::collections::HashMap;
+use std::env;
+use std::fs::{self, File};
+use std::io::Write;
+use std::net::IpAddr;
+use std::path::PathBuf;
+use std::str::FromStr;
+use std::sync::atomic::{AtomicU64, Ordering};
+use std::time::Duration;
+
+use rcgen::{
+    BasicConstraints, Certificate, CertificateParams, CertificateRevocationListParams,
+    DistinguishedName, DnType, ExtendedKeyUsagePurpose, Ia5String, IsCa, KeyIdMethod, KeyPair,
+    KeyUsagePurpose, RevocationReason, RevokedCertParams, RsaKeySize, SanType, SerialNumber,
+    SignatureAlgorithm, PKCS_ECDSA_P256_SHA256, PKCS_ECDSA_P384_SHA384, PKCS_ECDSA_P521_SHA512,
+    PKCS_ED25519, PKCS_RSA_SHA256, PKCS_RSA_SHA384, PKCS_RSA_SHA512,
+};
+use time::OffsetDateTime;
+
+fn main() {
```

Comment:
Maybe make this yield `Result<(), rcgen::Error>` so we can `?` all the errors? The try operator is quite a bit less noisy than `.unwrap()`...

(`Result<(), Box<dyn std::error::Error>>` also works if we need to unwrap other errors often.)

Reply:
Done, with the exception of `Option` `unwrap()`s where I didn't want to have to invent an error to map the `None` case to.